app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/PipelineDetailsActionsButton/index.tsx

```diff
@@ -73,65 +76,73 @@ const sanitizeConfig = (pipeline) => {
   return pipelineClone;
 };
 
-export default class PipelineDetailsActionsButton extends Component {
-  static propTypes = {
-    pipelineName: PropTypes.string,
-    description: PropTypes.string,
-    artifact: PropTypes.object,
-    config: PropTypes.object,
-    version: PropTypes.string,
-    lifecycleManagementEditEnabled: PropTypes.bool,
-    sourceControlManagementEnabled: PropTypes.bool,
-    editDraftId: PropTypes.string,
-    isLatestVersion: PropTypes.bool,
-    showCommitModal: PropTypes.bool,
+interface IPipelineDetailsActionsButtonProps {
+  pipelineName: string;
+  description: string;
+  artifact: any;
+  config: any;
+  version: string;
+  lifecycleManagementEditEnabled?: boolean;
+  sourceControlManagementEnabled?: boolean;
+  editDraftId?: string;
+  isLatestVersion?: boolean;
+  showCommitModal?: boolean;
+  pullLoading?: boolean;
+  pullStatus?: {
+    alertType: string;
+    message: string;
   };
+}
 
-  state = {
+class PipelineDetailsActionsButton extends Component<IPipelineDetailsActionsButtonProps> {
+  public state = {
     showExportModal: false,
     showDeleteConfirmationModal: false,
     showPopover: false,
     showDiscardConfirmation: false,
     showCommitModal: false,
+    loading: false,
+    deleteErrMsg: '',
+    extendedDeleteErrMsg: '',
   };
 
-  togglePopover = (showPopover = !this.state.showPopover) => {
+  public togglePopover = (showPopover = !this.state.showPopover) => {
```

Comment:
class methods are public by default

Reply:
the linting added those for me automatically